agents/packages/ethlisbon/skills/price_prophet/rounds.py

```diff
@@ -62,196 +65,110 @@ class SynchronizedData(BaseSynchronizedData):
     """
 
 
-class AnnotateDataRound(AbstractRound):
+class BaseRoundMixin:
+    """BaseRoundMixin"""
+
+    synchronized_data_class = SynchronizedData
+    done_event = Event.DONE
+    no_majority_event = Event.NO_MAJORITY
+
+
+class AnnotateDataRound(CollectSameUntilThresholdRound, BaseRoundMixin):
     """AnnotateDataRound"""
 
-    # TODO: replace AbstractRound with one of CollectDifferentUntilAllRound, CollectSameUntilAllRound, CollectSameUntilThresholdRound, CollectDifferentUntilThresholdRound, OnlyKeeperSendsRound, VotingRound
     round_id: str = "annotate_data"
     allowed_tx_type: Optional[TransactionType] = AnnotateDataPayload.transaction_type
     payload_attribute: str = "annotate_data"
-
-    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
-        """Process the end of the block."""
-        raise NotImplementedError
-
-    def check_payload(self, payload: AnnotateDataPayload) -> None:
-        """Check payload."""
-        raise NotImplementedError
-
-    def process_payload(self, payload: AnnotateDataPayload) -> None:
-        """Process payload."""
-        raise NotImplementedError
+    collection_key: str  # TODO
+    selection_key: str  # TODO
 
 
-class ModelValidationRound(AbstractRound):
+class ModelValidationRound(CollectSameUntilThresholdRound, BaseRoundMixin):
     """ModelValidationRound"""
 
-    # TODO: replace AbstractRound with one of CollectDifferentUntilAllRound, CollectSameUntilAllRound, CollectSameUntilThresholdRound, CollectDifferentUntilThresholdRound, OnlyKeeperSendsRound, VotingRound
     round_id: str = "model_validation"
     allowed_tx_type: Optional[TransactionType] = ModelValidationPayload.transaction_type
     payload_attribute: str = "model_validation"
-
-    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
-        """Process the end of the block."""
-        raise NotImplementedError
-
-    def check_payload(self, payload: ModelValidationPayload) -> None:
-        """Check payload."""
-        raise NotImplementedError
-
-    def process_payload(self, payload: ModelValidationPayload) -> None:
-        """Process payload."""
-        raise NotImplementedError
```

Comment:
do we not need these methods?

Reply:
- [`check_payload` and `process_payload`](https://github.com/valory-xyz/open-autonomy/blob/20551f16131c8de0144729f2422bcca91e2210b6/packages/valory/skills/abstract_round_abci/base.py#L1138-L1191) reside on the `CollectionRound` base class
- [`end_round`](https://github.com/valory-xyz/open-autonomy/blob/20551f16131c8de0144729f2422bcca91e2210b6/packages/valory/skills/abstract_round_abci/base.py#L1336-L1353) resides on `CollectSameUntilThresholdRound`

for a reference implementation, see e.g. [safe_deployment_abci]https://github.com/valory-xyz/open-autonomy/blob/main/packages/valory/skills/safe_deployment_abci/rounds.py